Description: Various changes to the upstream makefiles

* Don't strip binaries in Makefile.
* Install in /usr/games, not /usr/bin.
* Install cheats.bml into /usr/share/games/bsnes/, not ~/.config/bsnes.
* Name bsnes.png and bsnes.desktop explicitly.
* Include ALSA audio backend only on Linux.

Origin: http://anonscm.debian.org/gitweb/?p=pkg-games/bsnes.git;a=tree;f=debian/patches

Index: higan-092/higan/Makefile
===================================================================
--- higan-092.orig/higan/Makefile	2013-03-09 13:42:41.505530924 +0100
+++ higan-092/higan/Makefile	2013-03-09 14:01:04.781498105 +0100
@@ -16,8 +16,8 @@
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -I. -O3 -fomit-frame-pointer
-link    := -s
+flags   := -I. $(CFLAGS) $(CPPFLAGS)
+link    := $(LDFLAGS)
 objects := libco
 
 # profile-guided optimization mode
Index: higan-092/higan/target-ethos/Makefile
===================================================================
--- higan-092.orig/higan/target-ethos/Makefile	2013-01-12 22:59:47.000000000 +0100
+++ higan-092/higan/target-ethos/Makefile	2013-03-09 13:46:30.093524124 +0100
@@ -17,8 +17,11 @@
 # platform
 ifeq ($(platform),x)
   ruby := video.glx video.xv video.sdl
-  ruby += audio.alsa audio.openal audio.oss audio.pulseaudio audio.pulseaudiosimple audio.ao
+  ruby += audio.openal audio.oss audio.pulseaudio audio.pulseaudiosimple audio.ao
   ruby += input.sdl input.x
+  ifneq ($(findstring Linux,$(uname)),)
+    ruby += audio.alsa
+  endif
 else ifeq ($(platform),osx)
   ruby :=
   ruby += audio.openal
@@ -84,14 +87,11 @@
 	@echo Please do not run make install as root.
 	@echo The installer needs to know your home directory to install important files.
 else ifeq ($(platform),x)
-	sudo install -D -m 755 out/$(name) $(DESTDIR)$(prefix)/bin/$(name)
-	sudo install -D -m 644 data/$(name).png $(DESTDIR)$(prefix)/share/pixmaps/$(name).png
-	sudo install -D -m 644 data/$(name).desktop $(DESTDIR)$(prefix)/share/applications/$(name).desktop
-
-	mkdir -p ~/.config/$(name)
-	cp -R profile/* ~/.config/$(name)
-	cp data/cheats.bml ~/.config/$(name)/cheats.bml
-	chmod -R 777 ~/.config/$(name)
+	install -D -m 755 out/$(name) $(DESTDIR)$(prefix)/games/$(name)
+	install -D -m 644 data/higan.png $(DESTDIR)$(prefix)/share/pixmaps/higan.png
+	install -D -m 644 data/higan.desktop $(DESTDIR)$(prefix)/share/applications/higan.desktop
+	install -D -m 644 data/cheats.xml $(DESTDIR)$(prefix)/share/games/higan/cheats.xml
+	cp -R profile/* $(DESTDIR)$(prefix)/share/games/higan/profile/
 endif
 
 uninstall:
Index: higan-092/ananke/Makefile
===================================================================
--- higan-092.orig/ananke/Makefile	2013-01-13 00:18:14.000000000 +0100
+++ higan-092/ananke/Makefile	2013-03-09 14:03:14.661494241 +0100
@@ -2,7 +2,7 @@
 include phoenix/Makefile
 
 path := /usr/local/lib
-flags := -I. -O3 -fomit-frame-pointer
+flags := -I. $(CFLAGS) $(CPPFLAGS)
 ifeq ($(arch),win32)
   flags := -m32 $(flags)
 endif
